packages/firestore/test/integration/api/query.test.ts

```diff
@@ -200,11 +206,16 @@ apiDescribe('Queries', persistence => {
           ]);
           return storeEventFull.awaitEvent();
         })
-        .then(querySnap => {
+        .then(async querySnap => {
           expect(toDataArray(querySnap)).to.deep.equal([
             { v: 'a' },
             { v: 'b' }
           ]);
+          if (querySnap.metadata.fromCache) {
+            // We might receive an additional event if the first query snapshot
+            // was served from cache.
+            await storeEventFull.awaitEvent();
```

Comment:
I added this today as this test is pretty flaky when run as part of a larger test suite. It only ever fails with persistence enabled and seems to depend on whether listener 1 successfully received CURRENT before listener 2 is added.